scripts/gen-live-site/main.go

```diff
@@ -0,0 +1,271 @@
+// files with extension `.js` or `.css` are renamed to put a short sha into the name
+// for cache busting purposes (e.g. foo.css -> foo.1a2b3c4d.css).
+// We delay copying html files until after we have walked the src filetree and copy
+// them in a second pass afterwards so that we can update any references to renamed
+// files in them.
+// Return the skipped files (html) and a map of the files we renamed, or an error
+// if something went wrong.
+func copyTree(src, dest string) ([]string, map[string]string, error) {
+	skippedFiles := []string{}
+	renamedFiles := make(map[string]string)
+
+	srcfs := os.DirFS(src)
+	err := fs.WalkDir(srcfs, ".", func(filename string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		srcfile := filepath.Join(src, filename)
+		destfile := filepath.Join(dest, filename)
+		ext := filepath.Ext(filename)
+
+		switch mode := d.Type() & fs.ModeType; mode {
+		case fs.ModeDir:
+			return os.Mkdir(destfile, 0o777) //nolint:gosec // erroneous linter
+		case fs.ModeSymlink:
+			// We treat .js and .css files as normal files and make a
+			// copy, renaming the filename. This removes the symlink and
+			// duplicates the file. Otherwise the symlink is maintained as
+			// in in the dest tree.
+			if ext == ".js" || ext == ".css" {
+				break
+			}
+
+			// Would be nice to be able to ReadLink through a fs.FS:
+			// https://github.com/golang/go/issues/49580
+			l, err := os.Readlink(srcfile)
+			if err != nil {
+				return err
+			}
+			// Create the symlink
+			return os.Symlink(l, destfile)
+
+		case 0: // normal file
+			// handled below
+		default:
+			//nolint:goerr113 // dynamic errors in package main is ok
+			return fmt.Errorf("unknown file type: %s: %s", mode, srcfile)
+		}
+
+		switch ext {
+		case ".html":
+			skippedFiles = append(skippedFiles, filename)
+			return nil
+		case ".js", ".css":
+			shortSha, err := hashFile(srcfile)
+			if err != nil {
+				return err
+			}
+			basename := strings.TrimSuffix(filepath.Base(filename), ext)
+			target := basename + "." + shortSha + ext
+			destfile = filepath.Join(filepath.Dir(destfile), target)
+			if _, ok := renamedFiles[filename]; ok {
+				//nolint:goerr113 // dynamic errors in package main is ok
+				return fmt.Errorf("duplicate filename: %s", srcfile)
+			}
+			renamedFiles[filename] = target
+		}
+		return copyFile(srcfile, destfile)
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	return skippedFiles, renamedFiles, nil
+}
+
+func copyHTMLFiles(src, dest, domain string, skippedFiles []string, renamedFiles map[string]string) error {
+	for _, filename := range skippedFiles {
+		in, out, err := openInOut(filepath.Join(src, filename), filepath.Join(dest, filename))
+		if err != nil {
+			return err
+		}
+		defer in.Close() //nolint:errcheck // don't care about close failing on read-only files
+		err = updateHTMLFile(out, in, filename, domain, renamedFiles)
```

Comment:
before you return `in, out` and here you take `out, in` - on purpose?

Reply:
yeah sort of. the out, in is the "normal" order it seems (although I've always preferred the other way - but look at `io.Copy`. but `openOutIn` just felt wrong, but I can change it if you like

Reply:
no, I understand now. sounds good.

Reply:
ok, understand. let's leave as is.